src/search/search.component.ts

```diff
@@ -0,0 +1,223 @@
+import {
+	Component,
+	Input,
+	EventEmitter,
+	Output,
+	HostBinding
+} from "@angular/core";
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+import { I18n } from "../i18n/i18n.module";
+
+/**
+ * Used to emit changes performed on search components.
+ * @export
+ * @class SearchChange
+ */
+export class SearchChange {
+	/**
+	 * Contains the `Search` that has been changed.
+	 * @type {Search}
+	 * @memberof SearchChange
+	 */
+	source: Search;
+	/**
+	 * The value of the `Search` field encompassed in the `SearchChange` class.
+	 * @type {string}
+	 * @memberof SearchChange
+	 */
+	value: string;
+}
+
+/**
+ * @export
+ * @class Search
+ * @implements {ControlValueAccessor}
+ */
+@Component({
+	selector: "ibm-search",
+	template: `
+		<div
+			class="bx--search"
+			[ngClass]="{
+				'bx--search--sm': size === 'sm',
+				'bx--search--lg': size === 'lg',
+				'bx--search--light': theme === 'light'
+			}"
+			role="search">
+			<label class="bx--label" [for]="id">{{label}}</label>
+			<input
+				class="bx--search-input"
+				type="text"
+				role="search"
+				[id]="id"
+				[value]="value"
+				[placeholder]="placeholder"
+				[disabled]="disabled"
+				[required]="required"
+				(input)="onSearch($event.target.value)"/>
+			<svg
+				class="bx--search-magnifier"
+				width="16"
+				height="16"
+				viewBox="0 0 16 16">
+				<path
+					d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zm4.936-1.27l4.563 4.557-.707.708-4.563-4.558a6.5 6.5 0 1 1 .707-.707z"
+					fill-rule="nonzero"/>
+			</svg>
+			<button
+				class="bx--search-close"
+				[ngClass]="{
+					'bx--search-close--hidden': !value || value === 0
```

Comment:
`!value` should cover `value === 0` case. Is there a special reason why you need both? If not, do
```suggestion
					'bx--search-close--hidden': !value
```

Reply:
I actually meant `value.length === 0`.